runtime/entry-names.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Defines the macro RTNAME(n) which decorates the external name of a runtime
```

Comment:
I don't recall C runtime libraries have such name mangling; am I mis-remembering or else how do they get around version-specific naming?  (I'm thinking of compiler-support libraries, not the standard libraries.)

Reply:
The only examples that I know of use versioning at the library level and enforce it when dynamic libraries are loaded, leading to mysterious errors about GLIBC>=2.6 or something like that.

The naming scheme that I suggest doesn't depend on extra features in linkers, and exploits the fact that we control both sides of the interface.